src/Combinatorics/Graphs/functions.jl

```diff
@@ -1051,3 +1047,28 @@ Polyhedron in ambient dimension 6
 ```
 """
 fractional_matching_polytope(G::Graph{Undirected}) = polyhedron(Polymake.polytope.fractional_matching_polytope(pm_object(G)))
+
+
+################################################################################
+################################################################################
+##  Printing
+################################################################################
+################################################################################
+function Base.show(io::IO, ::MIME"text/plain", G::Graph{T}) where {T <: Union{Polymake.Directed, Polymake.Undirected}}
+  if nedges(G) > 0
+    println(io, "Graph{$T} with $(nvertices(G)) nodes and the following edges:")  # at least one new line is needed
+    for e in edges(G)
+      print(io, "($(src(e)), $(dst(e)))")
+    end
+  else
+    print(io, "Graph{$T} with $(nvertices(G)) nodes and no edges")
+  end
+end
+
+function Base.show(io::IO, G::Graph{T})  where {T <: Union{Polymake.Directed, Polymake.Undirected}}
+  if get(io, :supercompact, false)
+    print(io, "Graph{$T}")
```

Comment:
I am not objecting, just wondering: why not print it as "Direct graph" and "Undirected graph"?